File: danca/danca.py
def permutar_coluna(matriz, coluna1, coluna2):
    temp = []
    for i in matriz:
        temp.append(i[coluna1 - 1])
        i[coluna1-1] = i[coluna2-1]
        i[coluna2-1] = temp[matriz.index(i)]

def permutar_linha(matriz, linha1, linha2):
    matriz[linha1 - 1], matriz[linha2 - 1] = matriz[linha2 - 1], matriz[linha1 - 1]

N, M, P = map(int, input().split())

matriz = [[j + 1 + i * M for j in range(M)] for i in range(N)]

ordens = []
for i in range(P):
    a, b, c = input().split()
    ordens.append([a, int(b), int(c)])

for i in ordens:
    if i[0] == 'C':
        permutar_coluna(matriz, i[1], i[2])
    elif i[0] == 'L':
        permutar_linha(matriz, i[1], i[2])

for i in matriz:
    print(*i)

File: danca/danca.c
#include <stdio.h>
#include "../souls.h"

void permutar_linha(int N, int M, int matriz[N][M], int linha1, int linha2) {
    for (int i = 0; i < M; i++) {
        int temp[M];
        temp[i] = matriz[linha1 - 1][i];
        matriz[linha1 - 1][i] = matriz[linha2 - 1][i];
        matriz[linha2 - 1][i] = temp[i];
    }
}

void permutar_coluna(int N, int M, int matriz[N][M], int coluna1, int coluna2) {
    for (int i = 0; i < N; i++) {
        int temp[N];
        temp[i] = matriz[i][coluna2 - 1];
        matriz[i][coluna2 - 1] = matriz[i][coluna1 - 1];
        matriz[i][coluna1 - 1] = temp[i];
    }
}

int main() {
    int N, M, P;
    scanf("%d %d %d", &N, &M, &P);

    int matriz[N][M];
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < M; j++) {
            matriz[i][j] = i * M + j + 1;
        }
    }

    int ordens[P][3];
    char tipo;

    for (int i = 0; i < P; i++) {
        scanf(" %c %d %d", &tipo, &ordens[i][1], &ordens[i][2]);
        ordens[i][0] = tipo;
    }

    for (int i = 0; i < P; i++) {
        if (ordens[i][0] == 67) {
            permutar_coluna(N, M, matriz, ordens[i][1], ordens[i][2]);
        }
        else if (ordens[i][0] == 76) {
            permutar_linha(N, M, matriz, ordens[i][1], ordens[i][2]);
        }
    }

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < M; j++) {
            printf("%d ", matriz[i][j]);
        }
        printf("\n");
    }

    return 0;
}
